addons/website_sale/models/product.py

```diff
@@ -179,7 +179,10 @@ def _is_quick_add_to_cart_possible(self, parent_combination=None):
 
         if not self._is_add_to_cart_possible(parent_combination):
             return False
-        if len(self._get_possible_variants(parent_combination)) != 1:
+        # Ideally the following condition should test for _get_possible_variants
+        # but it is too slow to call for multiple products, so we just check
+        # for all the variants (possible or not).
+        if len(self.product_variant_ids) != 1:
```

Comment:
Better

- if len(self.`with_prefetch()`.product_variant_ids) != 1 


(Not sure if `>` will be faster or not instaed of `!=` but `>` is used in base already for similar condition)

- or we can use field `product_variant_count` 


https://github.com/odoo/odoo/blob/12.0/addons/product/models/product_template.py#L123


What do you think on this ?

Reply:
It is actually still a problem (this and also my suggestion solution) because after thinking about it we shouldn't try to read all of the variants, just know if there are at least 2 possible.

I'm currently trying to update _get_first_possible_combination into _get_nth_possible_combination, so use it as a way to know if there are at least 2 without having to read any of them.

Reply:
Will  `with_context(prefetch_fields=False)`   make sense here ? (Not sure about this)

Reply:
I appreciate the feedback. I think you are correct that disabling `prefetch` would be good if I wanted to keep using `product_variant_ids` for this, or even using `product_variant_count`.

But I think it is better to keep working with the combinations instead of the variants like I did for the rest of the configurator fixes, and so in priority I want to try to not use `product_variant_ids` if I find a way to not use it.